src/main/java/temporal/SimpleWorkflowImpl.java

```diff
@@ -0,0 +1,77 @@
+package temporal;
+
+import io.temporal.activity.ActivityOptions;
+import io.temporal.api.enums.v1.ParentClosePolicy;
+import io.temporal.workflow.Async;
+import io.temporal.workflow.ChildWorkflowOptions;
+import io.temporal.workflow.Workflow;
+import java.time.Duration;
+import java.util.LinkedList;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import temporal.model.OrderInfo;
+
+public class SimpleWorkflowImpl implements SimpleWorkflow {
+
+  private boolean exit = false;
+
+  private static final Logger logger = LoggerFactory.getLogger(SimpleWorkflowImpl.class);
+
+  private final List<OrderInfo> registeredOrderInfos = new LinkedList<>();
+
+  private final OrderProcessingActivity orderProcessingActivity =
+      Workflow.newActivityStub(
+          OrderProcessingActivity.class,
+          ActivityOptions.newBuilder().setStartToCloseTimeout(Duration.ofSeconds(2)).build());
+
+  private io.temporal.workflow.WorkflowInfo info;
+
+  @Override
+  public void update(OrderInfo orderInfo) {
+    logger.info("Updating order in Workflow: " + orderInfo.toString());
+    // Add new order updates to the queue
+    orderProcessingActivity.update(orderInfo);
+  }
+
+  @Override
+  public void register(OrderInfo orderInfo) {
+    logger.info("Registering order in Workflow: " + orderInfo.toString());
+    this.registeredOrderInfos.add(orderInfo);
+  }
+
+  @Override
+  public void notifyCustomer(OrderInfo orderInfo) {
+
+    logger.info("Notifying customer for order from parent workflow: " + orderInfo.toString());
```

Comment:
Rather than string concatenation, it is recommended to use the string parameterization format for log statements

Reply:
@reselbob You do not need to use String.format explicitly. slf4j loggers will substitute values for the {} placeholders in the parameterized string template

Reply:
addressed

Reply:
OK